### PR TITLE
Move callsite! macro out of core

### PR DESCRIPTION
**Pull request by hawkw**
Depends on #85.

Since #83, callsites no longer need to be constructed by the `tokio-trace-core`
crate; they must simply implement the `Callsite` trait. Thus, the `callsite`
macro need lo longer be in `-core`. This branch moves it to the `tokio-trace`
crate.

Additionally, I've replaced the `#[doc(hidden)]` `Kind` enum with a stable
`MetaKind` struct. The `callsite!` macro no longer depends on any
`#[doc(hidden)]` APIs; the remaining uses of `#[doc(hidden)]` are in `Value`
code that will soon be rewritten.

Signed-off-by: Eliza Weisman <eliza@buoyant.io>

### CODE REVIEW

File: tokio-trace-core/src/lib.rs [hunk 1]
```diff
@@ -244,6 +176,10 @@ pub struct Event<'a> {
 /// [`Subscriber`]: ::Subscriber
 #[derive(Clone, Debug, Eq, Hash)]
 pub struct Meta<'a> {
+    // TODO: The fields on this type are currently `pub` because it must be able
+    // to be constructed statically by macros. However, when `const fn`s are
+    // available on stable Rust, this will no longer be necessary. Thus, these
+    // fields should be made private when `const fn` is stable.
```
Comment (davidbarsky) [inline tokio-trace-core/src/lib.rs]: Making these fields private would be a breaking change, correct?

Comment (hawkw) [reply tokio-trace-core/src/lib.rs]: Yes. In the context I think it's probably worth it.

Comment (davidbarsky) [reply tokio-trace-core/src/lib.rs]: I agree, just wanted to ask about that.

Comment (hawkw) [reply tokio-trace-core/src/lib.rs]: 👍

**Pull request merged**